README.md

```diff
@@ -1,70 +1,75 @@
-    - If you need to change the app/host name, please make sure to edit the host ```Makefile```, and change the variable ```APP_U```.
+### Environment
 
-## Build your project
+**NOTE:** If you opted to install `gcc` as your compiler, make sure you run
+```
+export CC=gcc; export CXX=g++
+```
+Otherwise you may safely continue.
 
-### Before starting the building process, please make sure you downloaded the Rust SGX SDK repository, we're going to need the EDL and headers files joined in the SDK.
+The Rust and Intel SGX SDKs need to be installed and the
+relevant environment variables need to be set.  In order to facilitate this, we
+use the convenience scripts provided at [rust-sgx-sdk-env].
 
-Once you downloaded the Rust SGX SDK, you have multiple ways to start the building process: 
-- Run this command: ```CUSTOM_EDL_PATH=~/teaenclave/edl CUSTOM_COMMON_PATH=~/teaenclave/common make``` (replace ```~/teaenclave``` by the actual SDK location)
-- You can also run the command export (```export CUSTOM_EDL_PATH=~/teaenclave/edl```), and specify the variables before calling make. It is advised to add this command on your ```.bashrc``` file (if you use bash), or your favorite shell configuration file.
+1. Make sure the `docker` daemon is running, otherwise start it with
+   ```
+   $ systemctl start snap.docker.dockerd
+   ```
+2. In order to run the scripts as a non-root user, follow the
+   [docker-postinstall](post-installation instructions) set out in the Docker
+   documentation (note, in particular, that a restart may be necessary).
+3. Run the latest "prepare" script:
+   ```
+   $ ./prepare-1.1.14-intel-2.15.1.sh
+   ```
+4. Finally, assuming `bash` is the current shell, source the environment file in
+   the top level of the repository:
+   ```
+   $ source environment
+   ```
 
-### By default, your project will be compiled in hardware mode. If you wish to compile your project in software/simulation mode, you will need to specify it, either by adding ```SGX_MODE=SW``` before make, or by setting the SGX_MODE variable environment to SW.
+### Instructions
 
-### Cargo is used by default when compiling, but you can also use Xargo either by adding ```XARGO_SGX=1``` before make, or by setting the XARGO_SGX variable environment to 1. You will also need to specify Xargo library path with XARGO_PATH.
+1. Before proceeding, make sure your [environment is set up](#environment)
+   properly.
+2. Clone the project repository into the current directory
+    ```
+    $ git clone https://github.com/ntls-io/wasm-exec-sgx
+    ```
+   and `cd` into it.
+3. Run `make all` to compile the entire project.
+4. In order to test the provided Wasm binary, change to the current directory
+   to the `wasmi-impl` subdirectory and execute the following:
+    ```
+    cargo test
+    ```
 
```

Comment:
Let's add another bullet point or make this point 4 and make point 4 -> point 5:

5. To run the main application, change to `bin/` and execute the `app`:
   ```
   ./app
   ```

Reply:
Would you mind providing a diff via the GH "suggestion" feature? @billguo99

Reply:
Cool, done